### PR TITLE
Set hostname to workspace ID rather than instance ID

### LINKED ISSUE #7359: Set hostname to workspace ID rather than instance ID
Right now we're inheriting the hostname set by Kubernetes. Kubernetes uses the pod name as hostname.
Instead of bind-mounting `/etc/hostname` we could just write our own file which contains the workspace name.

Benefits: looks much nicer when folks start customising their `PS1`, or when using tailscale.

### PR DESCRIPTION
**Pull request by iQQBot**
## Description

Right now we're inheriting the hostname set by Kubernetes. Kubernetes uses the pod name as hostname.
We can change pod.spec.hostname to overwrite it

## Related Issue(s)

Fixes #7359

## How to test

1. start a workspace
2. run `cat /etc/hostname` and `cat /proc/sys/kernel/hostname`
3. run `hostname` see hostname set to workspace name instead of workspace instance id

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **aledbf** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @aledbf` in a comment when ready.

Associated issue: *#7359*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by csweichel — APPROVED** [review]
LGTM. Thanks @iQQBot :)

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 61094197e127973483dd85274f3c5c12fd09c81f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7365#pullrequestreview-841447382" title="Approved">csweichel</a>*

Associated issue: *#7359*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
